gadget/install/install.go

```diff
@@ -510,6 +544,104 @@ func WriteContent(onVolumes map[string]*gadget.Volume, observer gadget.ContentOb
 	return onDiskVols, nil
 }
 
+func FinishEncryption(model gadget.Model, setupData *EncryptionSetupData) error {
+	// Save storage traits to save/data partitions
+	encryptionParams := map[string]gadget.StructureEncryptionParameters{}
+	var volName string
+	for name, p := range setupData.Parts {
+		encryptionParams[name] = p.encryptionParams
+		volName = p.volName
+	}
+	optsPerVol := map[string]*gadget.DiskVolumeValidationOptions{
+		// this assumes that the encrypted partitions above are always only on the
+		// system-boot volume, this assumption may change
+		volName: {
+			ExpectedStructureEncryption: encryptionParams,
+		},
+	}
```

Comment:
are we sure there is not more data that is different in the case of encryption? I suppose the we need to compare saved traits with the uc20 case and also add some traits checks to the spread test?